### PR TITLE
feat: create page and table action quick actions for OData(v4) applications

### PR DESCRIPTION
**Pull request by nahmed22**
Enhance Quick action list to add custom actions (OData v4)

### CODE REVIEW

File: packages/adp-tooling/templates/rta/common/v4-table-action.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<core:FragmentDefinition  xmlns:core='sap.ui.core' xmlns='sap.m'>
+   <actiontoolbar:ActionToolbarAction xmlns:actiontoolbar="sap.ui.mdc.actiontoolbar" id="<%- ids.customToolbarAction%>" >
+        <Button xmlns:m="sap.m" id="<%- ids.customActionButton %>" visible="true" text="Click here" />
```
Comment (voicis) [inline packages/adp-tooling/templates/rta/common/v4-table-action.xml:3]: Initially I thought that the Quick Action did not work, but then it turned out that the action name was just not clear enough to indicate that it was just added. We should use a text for the button that is a bit more distinctive and can be easier to understand that it is something that was just added.

```suggestion
        <Button xmlns:m="sap.m" id="<%- ids.customActionButton %>" visible="true" text="New Action" />
```

File: .changeset/mighty-timers-turn.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@sap-ux/adp-tooling": patch
+"@sap-ux-private/preview-middleware-client": patch
```
[the remark this answers is not in the document]
Comment (nahmed22) [reply .changeset/mighty-timers-turn.md:3]: done

[the remark this answers is not in the document]
Comment (nahmed22) [reply .changeset/mighty-timers-turn.md:3]: added

**Review by voicis — APPROVED** [review]
- Code changes looks good
- Good test coverage
- Tested locally

**Review by nikmace — APPROVED** [review]
Code looks good
Coverage is good
Did not test manually

**Pull request merged**